plans/testground-test/signalbarrier.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/ipfs/testground/sdk/sync"
+)
+
+// This function is intended to exercise the barier time
+// This makes sense only when there is a lot of nodes.
+func signalBarrierBench(runenv *runtime.RunEnv) error {
+	ctx := context.Background()
+
+	watcher, writer := sync.MustWatcherWriter(ctx, runenv)
+	iterations := runenv.IntParam("iterations")
+
+	barrierMetric := runtime.MetricDefinition{
+		Name:           "Barrier Latency",
+		Unit:           "nanoseconds",
+		ImprovementDir: -1,
+	}
+
+	// Wait until all the nodes are started
+	runenv.RecordMessage("waiting for others to become ready.")
+	writer.SignalEntry(ctx, sync.State("ready"))
+	_ = <-watcher.Barrier(ctx, sync.State("ready"), int64(runenv.TestInstanceCount))
+	runenv.RecordMessage("Ready to begin.")
+
+	var total int64
+
+	// ping-pong back and forth, with a barrier before each ping
+	for i := 1; i <= iterations; i += 1 {
+		ping := sync.State("ping")
+		pong := sync.State("pong")
+		start := time.Now().UnixNano()
+		writer.SignalEntry(ctx, ping)
+		_ = <-watcher.Barrier(ctx, ping, int64(i*runenv.TestInstanceCount))
+		end := time.Now().UnixNano()
+		writer.SignalEntry(ctx, pong)
+		nanoseconds := end - start
+		total = nanoseconds
+		runenv.RecordMetric(&barrierMetric, float64(nanoseconds))
+	}
+	runenv.RecordMessage("Average barrier latency, %d", int(total)/iterations)
```

Comment:
This test does exactly what we need, but also showcases that we need the metrics library - the average is not a good enough metric to tell us if our system is performing well - I think we'd want to measure:
- 95%, 99%, max for both: a) writer.signalEntry and b) watcher.Barrier;